src/command/cmd_funcs.c

```diff
@@ -9296,33 +9296,74 @@ cmd_mam(ProfWin* window, const char* const command, gchar** args)
 gboolean
 cmd_editor(ProfWin* window, const char* const command, gchar** args)
 {
-    const char* filename = "/tmp/profanity.txt";
+    /* Build temp file name */
+    GString * tempfile = g_string_new(g_get_tmp_dir());
+    g_string_append(tempfile, "/profanity-");
+    xmpp_ctx_t* const ctx = connection_get_ctx();
+    if(!ctx) {
+        cons_show("Editor: Not connection");
+        return TRUE;
+    }
+    char* uuid = xmpp_uuid_gen(ctx);
+    g_string_append(tempfile, uuid);
+    if (uuid) {
+        xmpp_free(ctx, uuid);
+    }
+    g_string_append(tempfile, ".txt");
+
+    // tempfile should be something like
+    // /tmp/profanity-f2f271dd-98c8-4118-8d47-3bd49c8e2e63.txt
+    const char* filename = tempfile->str;
+
+    // Check if file exists and create file
```

Comment:
Is it really appropriate to exit if the file exists? I'm not sure whether actually we would just open it as is or even rewrite? Is there a reason why we need several such files even? Maybe we should just always write to `/tmp/profanity-externaledit` or something? If a person chooses to use the editor command often it could mean he creates plenty of files in /tmp. And if this is on a server or somewhere where profanity runs for weeks it could mean that it uses lots of space. If the system is configured to only clean /tmp upon restart that will mean wasting plenty of space for a long time on a server for example.

Reply:
Just now saw that you delete the file at the end. So the space thing doesn't make sense. But makes me wonder even more why we need a random filename :)

Reply:
PM from @StefanKropp "we need random in case multiple instanced of profanity are running"

Reply:
Too bad, continuing typing would be nice. But let that be a feature request later.

Reply:
> Too bad, continuing typing would be nice. But let that be a feature request later.
I think we can think about more features. Maybe /editor <file> also /editor correct to use the message correction in editor mode,.... we will see :-)